pkg/raft/opts.go

```diff
@@ -144,7 +170,21 @@ func NodeUrl(addr string) Option {
 	return func(opts *options) error {
 		addr = strings.TrimSpace(addr)
 		if addr == "" {
-			return errors.New("Nexus listen address must be given")
+			//try to auto determine nodeUrl from clusterUrl
+			localIps, err := getLocalIPAddress()
+			if err == nil {
+				for _, clusterUrl := range opts.clusterUrls {
+					//check if localIps contains this
+					host, _, _ := net.SplitHostPort(clusterUrl.Host)
```

Comment:
Any reason we're not checking for the error value returned ?

Reply:
Really doesn't matter. The type of host is string, so even if error happens, the value of host would be empty, and the next lookup would fail, eventually, the last error would be returned.